core/Db/Schema/Tidb.php

```diff
@@ -27,6 +25,18 @@ public function supportsComplexColumnUpdates(): bool
         return false;
     }
 
+    public function getDefaultCollationForCharset(string $charset): string
+    {
+        $collation = parent::getDefaultCollationForCharset($charset);
+
+        if ($charset === 'utf8mb4' && $collation = 'utf8mb4_bin') {
```

Comment:
Use yoda style comparison or
```suggestion
        if ($charset === 'utf8mb4' && $collation === 'utf8mb4_bin') {
```